frontend/packages/dev-console/src/components/catalog/PinnedResourcesConfiguration.tsx

```diff
@@ -0,0 +1,340 @@
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { connect } from 'react-redux';
+import {
+  DiscoveryResources,
+  ExtensionK8sModel,
+  K8sKind,
+  K8sModel,
+  ResourceIcon,
+} from '@console/dynamic-plugin-sdk/src/lib-core';
+import {
+  K8sResourceKind,
+  modelFor,
+  referenceForGroupVersionKind,
+  referenceForModel,
+} from '@console/internal/module/k8s';
+import { RootState } from '@console/internal/redux';
+import { YellowExclamationTriangleIcon, useTelemetry } from '@console/shared';
+import {
+  useDebounceCallback,
+  useConsoleOperatorConfig,
+  LoadError,
+  SaveStatus,
+  SaveStatusProps,
+  patchConsoleOperatorConfig,
+} from '@console/shared/src/components/cluster-configuration';
+import {
+  Perspective,
+  PerspectivePinnedResource,
+  PerspectiveVisibilityState,
+  usePerspectives,
+} from '@console/shared/src/hooks/perspective-utils';
+import './PinnedResourcesConfiguration.scss';
+
+// Blacklist known duplicate resources.
+const blacklistGroups = ImmutableSet([
+  // Prefer rbac.authorization.k8s.io/v1, which has the same resources.
+  'authorization.openshift.io',
+]);
+
+const blacklistResources = ImmutableSet([
+  // Prefer core/v1
+  'events.k8s.io/v1beta1.Event',
+]);
+
+type PerspectivesConsoleConfig = K8sResourceKind & {
+  spec: {
+    customization?: {
+      perspectives: Perspective[];
+    };
+  };
+};
+
+type DefaultPins = {
+  dev?: ExtensionK8sModel[];
+};
+
+type PinnedResourcesConfigurationProps = {
+  readonly: boolean;
+  groupVersionMap: DiscoveryResources['groupVersionMap'];
+  allK8sModels: ImmutableMap<string, K8sModel>;
+};
+
+const PinnedResourcesConfiguration: React.FC<PinnedResourcesConfigurationProps> = ({
+  readonly,
+  allK8sModels,
+  groupVersionMap,
+}) => {
+  const { t } = useTranslation();
+  const fireTelemetryEvent = useTelemetry();
+  const perspectiveExtensions = usePerspectives();
+  const [pinnedResources, setPinnedResources] = React.useState<PerspectivePinnedResource[]>();
+  const [perspectiveData, setPerspectiveData] = React.useState<Perspective[]>();
+  const [pinnedResourcesConfigured, setPinnedResourcesConfigured] = React.useState<
+    PerspectivePinnedResource[]
+  >();
+  const defaultPins: DefaultPins = React.useMemo(
+    () =>
+      perspectiveExtensions.reduce(
+        (acc, e) => ({
+          ...acc,
+          [e.properties.id]: (e.properties.defaultPins || []).map((gvk) => gvk),
+        }),
+        {},
+      ),
+    [perspectiveExtensions],
+  );
+
+  const resources = React.useMemo(() => {
+    return allK8sModels
+      ?.filter(({ apiGroup, apiVersion, kind, verbs }) => {
+        // Remove blacklisted items.
+        if (
+          blacklistGroups.has(apiGroup) ||
+          blacklistResources.has(`${apiGroup}/${apiVersion}.${kind}`)
+        ) {
+          return false;
+        }
+
+        // Only show resources that can be listed.
+        if (!_.isEmpty(verbs) && !_.includes(verbs, 'list')) {
```

Comment:
Isn't this the same without lodash?

```suggestion
        if (!verbs?.some(v => v === 'list')) {
```